ezsystems/ezplatform-richtext
# EZP-31868: improve richtext embed access check on 404 page

| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31868](https://jira.ez.no/browse/EZP-31868)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | 1.1
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

1) Render richtext field in pagelayout (for example, in footer).
2) Put embed image in the richtext field.
3) Open valid URL to verify anonymous has access to the image.
3) Open some 404 URL

AR:
500 error returned after max nested level reached

ER:
404 Page properly displayed

Developer notes:
It's because there is no firewall for 404 page, and there is no token in token storage service.
So, it threw "AuthenticationCredentialsNotFoundException".
Subrequest tries to render an error page, which extends pagelayout, which tries to render the same richtext field, and it leads to an infinitive loop.

**TODO**:
